dns-rc-autoscaler/nanny/kubernetes_client.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nanny
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+
+	api "k8s.io/kubernetes/pkg/api"
+	client "k8s.io/kubernetes/pkg/client/clientset_generated/release_1_3"
+)
+
+const replicationControllerKind = "ReplicationController"
+const createdByAnnotationName = "kubernetes.io/created-by"
+
+type kubernetesClient struct {
+	namespace, pod string
+	clientset      *client.Clientset
+}
+
+// Find this pod's owner references, find the replication controller parent and return its name
+// This is acquired from the pod's annotation "kubernetes.io/created-by"
+//  annotations:
+//    kubernetes.io/created-by: |
+//      {"kind":"SerializedReference","apiVersion":"v1","reference":
+//      {"kind":"ReplicationController","namespace":"default",
+//      "name":"pod-autoscaler-nanny",
+//      "uid":"7ccf55b9-5073-11e6-8321-42010af00002",
+//      "apiVersion":"v1",
+//      "resourceVersion":"20339"}}
+//
+
+func (k *kubernetesClient) GetParentRc(namespace, podname string) (string, error) {
+	pod, err := k.clientset.CoreClient.Pods(namespace).Get(podname)
+	if err != nil {
+		return "", err
+	}
+	createdBy, ok := pod.ObjectMeta.Annotations[createdByAnnotationName]
+	if !ok || len(createdBy) == 0 {
+		// Empty or missing created-by annotation would mean this is a
+		// standalone pod and does not have an RC/RS managing it
+		return "", fmt.Errorf("Pod does not have a created-by annotation")
+	}
+	type RCInner struct {
+		Kind string `json:"kind"`
+		Name string `json:"name"`
+	}
+	type AnnotationT struct {
+		Reference RCInner `json:"reference"`
+	}
+	var annot AnnotationT
+	err = json.Unmarshal([]byte(createdBy), &annot)
```

Comment:
Isn't this annotation already a serialized ObjectReference?  Is that type not exposed by our client?